dotnet/src/SemanticKernel.Core/Memory/Mem0/Mem0Behavior.cs

```diff
@@ -0,0 +1,178 @@
+        this._threadId = options?.ThreadId;
+        this._userId = options?.UserId;
+        this._scopeToPerOperationThreadId = options?.ScopeToPerOperationThreadId ?? false;
+        this._contextPrompt = options?.ContextPrompt ?? DefaultContextPrompt;
+
+        this._mem0Client = new(httpClient);
+    }
+
+    /// <inheritdoc/>
+    public override Task OnThreadCreatedAsync(string? threadId, CancellationToken cancellationToken = default)
+    {
+        this.ValidatePerOperationThreadId(threadId);
+
+        this._perOperationThreadId ??= threadId;
+        return Task.CompletedTask;
+    }
+
+    /// <inheritdoc/>
+    public override async Task OnNewMessageAsync(string? threadId, ChatMessage newMessage, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(newMessage);
+        this.ValidatePerOperationThreadId(threadId);
+
+        switch (newMessage.Role)
+        {
+            case ChatRole u when u == ChatRole.User:
+            case ChatRole a when a == ChatRole.Assistant:
+            case ChatRole s when s == ChatRole.System:
+                break;
+            default:
+                return;
+        }
+
+        this._perOperationThreadId ??= threadId;
+
+        if (!string.IsNullOrWhiteSpace(newMessage.Text))
+        {
+            await this._mem0Client.CreateMemoryAsync(
+                this._applicationId,
+                this._agentId,
+                this._scopeToPerOperationThreadId ? this._perOperationThreadId : this._threadId,
+                this._userId,
+                newMessage.Text,
+                newMessage.Role.Value).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc/>
+    public override async Task<AIContextPart> OnModelInvokeAsync(ICollection<ChatMessage> newMessages, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(newMessages);
+
+        string inputText = string.Join(
+            Environment.NewLine,
+            newMessages.
+                Where(m => m is not null && !string.IsNullOrWhiteSpace(m.Text)).
+                Select(m => m.Text));
+
+        var memories = await this._mem0Client.SearchAsync(
+                this._applicationId,
+                this._agentId,
+                this._scopeToPerOperationThreadId ? this._perOperationThreadId : this._threadId,
+                this._userId,
+                inputText).ConfigureAwait(false);
+
+        var lineSeparatedMemories = string.Join(Environment.NewLine, memories);
+        return new()
+        {
+            Instructions =
+                $"""
+                {this._contextPrompt}
+                {lineSeparatedMemories}
+                """
+        };
+    }
+
+    /// <summary>
+    /// Plugin method to clear memories for the current agent/thread/user.
+    /// </summary>
+    /// <returns>A task that completes when the memory is cleared.</returns>
+    public async Task ClearStoredMemoriesAsync()
+    {
+        await this._mem0Client.ClearMemoryAsync(
```

Comment:
Direct delegation of task is possible here.

```suggestion
    public Task ClearStoredMemoriesAsync()
    {
        return this._mem0Client.ClearMemoryAsync(
```

Reply:
Thanks, updated.